diff --git a/node_modules/typescript/lib/tsc.js b/node_modules/typescript/lib/tsc.js
index 636e39a..04ad911 100644
--- a/node_modules/typescript/lib/tsc.js
+++ b/node_modules/typescript/lib/tsc.js
@@ -81030,7 +81030,7 @@ var ts;
             var result = factory.createParenthesizedExpression(factory.createCallExpression(outer, undefined, extendsClauseElement
                 ? [ts.visitNode(extendsClauseElement.expression, visitor, ts.isExpression)]
                 : []));
-            ts.addSyntheticLeadingComment(result, 3, "* @class ");
+            ts.addSyntheticLeadingComment(result, 3, "#__PURE__");
             return result;
         }
         function transformClassBody(node, extendsClauseElement) {
diff --git a/node_modules/typescript/lib/tsserver.js b/node_modules/typescript/lib/tsserver.js
index 60a159a..c3f428e 100644
--- a/node_modules/typescript/lib/tsserver.js
+++ b/node_modules/typescript/lib/tsserver.js
@@ -97080,7 +97080,7 @@ var ts;
             /*typeArguments*/ undefined, extendsClauseElement
                 ? [ts.visitNode(extendsClauseElement.expression, visitor, ts.isExpression)]
                 : []));
-            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "* @class ");
+            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "#__PURE__");
             return result;
         }
         /**
diff --git a/node_modules/typescript/lib/tsserverlibrary.js b/node_modules/typescript/lib/tsserverlibrary.js
index fcb5898..4075ffb 100644
--- a/node_modules/typescript/lib/tsserverlibrary.js
+++ b/node_modules/typescript/lib/tsserverlibrary.js
@@ -97274,7 +97274,7 @@ var ts;
             /*typeArguments*/ undefined, extendsClauseElement
                 ? [ts.visitNode(extendsClauseElement.expression, visitor, ts.isExpression)]
                 : []));
-            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "* @class ");
+            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "#__PURE__");
             return result;
         }
         /**
diff --git a/node_modules/typescript/lib/typescript.js b/node_modules/typescript/lib/typescript.js
index 3be91a8..5e77e96 100644
--- a/node_modules/typescript/lib/typescript.js
+++ b/node_modules/typescript/lib/typescript.js
@@ -97274,7 +97274,7 @@ var ts;
             /*typeArguments*/ undefined, extendsClauseElement
                 ? [ts.visitNode(extendsClauseElement.expression, visitor, ts.isExpression)]
                 : []));
-            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "* @class ");
+            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "#__PURE__");
             return result;
         }
         /**
diff --git a/node_modules/typescript/lib/typescriptServices.js b/node_modules/typescript/lib/typescriptServices.js
index e958d4e..e103372 100644
--- a/node_modules/typescript/lib/typescriptServices.js
+++ b/node_modules/typescript/lib/typescriptServices.js
@@ -97274,7 +97274,7 @@ var ts;
             /*typeArguments*/ undefined, extendsClauseElement
                 ? [ts.visitNode(extendsClauseElement.expression, visitor, ts.isExpression)]
                 : []));
-            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "* @class ");
+            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "#__PURE__");
             return result;
         }
         /**
diff --git a/node_modules/typescript/lib/typingsInstaller.js b/node_modules/typescript/lib/typingsInstaller.js
index 715445f..ea0d2d8 100644
--- a/node_modules/typescript/lib/typingsInstaller.js
+++ b/node_modules/typescript/lib/typingsInstaller.js
@@ -97069,7 +97069,7 @@ var ts;
             /*typeArguments*/ undefined, extendsClauseElement
                 ? [ts.visitNode(extendsClauseElement.expression, visitor, ts.isExpression)]
                 : []));
-            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "* @class ");
+            ts.addSyntheticLeadingComment(result, 3 /* MultiLineCommentTrivia */, "#__PURE__");
             return result;
         }
         /**
